src/keas/kmi/facility.py

```diff
@@ -100,7 +106,9 @@ def encrypt_file(self, key, fsrc, fdst, chunksize=24*1024):
         encryptionKey = self._bytesToKey(self.getEncryptionKey(key))
 
         # 2. Create a random initialization vector
-        iv = ''.join(chr(random.randint(0, 0xFF)) for i in range(16))
+        # bytes(bytearray(generator)) is needed for Python 2,
+        # with Python 3 bytes(generator) works
+        iv = bytes(bytearray((random.randint(0, 0xFF)) for i in range(16)))
```

Comment:
(If this was written today, I'd suggest using [`cryptography`](https://cryptography.io/en/latest/) instead of M2Crypto.)